### PR TITLE
remove DM

### PR DESCRIPTION
**Pull request by jgrund**
Currently device mapper block devices are detected by parsing _dmsetup table_ command output rather than deriving information from the **device_scanner** output.

Integrate detection of device mapper block devices into **block_devices** module using output from the device scanner in order to move towards an event driven device discovery architecture.

The **device_mapper**  module is not needed to detect device mapper block devices but remains because elements are used in other remaining code and legacy tests (to be removed subsequently).

Changes include:
- relocate normalised device table into block devices
- remove unused 'mpath' attribute returned by linux agent device plug-in
- link dm object slave devices to relevant device mapper path using mm as lookup
- parse lvm uuid into vg/lv uuids
- add _normalised device table_ mappings for mpath devices
- remove legacy tests including those for nonlocal lvm devices
- update tests to cover lvm striped and linear devices, mpath configurations and device mapper devices

 Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

This change is [Reviewable](https://reviewable.io/reviews/intel-hpdd/intel-manager-for-lustre/337)

### CODE REVIEW

File: chroma-agent/chroma_agent/device_plugins/linux_components/block_devices.py [hunk 1]
```diff
@@ -94,7 +95,13 @@ def as_device(x):
         'device_path': x.get('DEVPATH'),
         'partition_number': x.get('ID_PART_ENTRY_NUMBER'),
         'is_ro': x.get('IML_IS_RO'),
-        'parent': None
+        'parent': None,
+        'dm_multipath': x.get('DM_MULTIPATH_DEVICE_PATH'),
+        'dm_lv': x.get('DM_LV_NAME'),
+        'dm_vg': x.get('DM_VG_NAME'),
+        'dm_uuid': x.get('DM_UUID'),
```
Comment (jgrund) [inline chroma-agent/chroma_agent/device_plugins/linux_components/block_devices.py]: Could this be None?

File: chroma-agent/chroma_agent/device_plugins/linux_components/block_devices.py [hunk 2]
```diff
@@ -134,33 +141,142 @@ def fetch_device_list():
...
-            ndp.add_normalized_device(x, y)
+def parse_lvm_uuids(dm_uuid):
+    """
+    :param dm_uuid: device mapper uuid string (combined vg and lv with lvm prefix) from device_scanner
+    :return: tuple of vg and lv UUIDs
+    """
+    lvm_pfix = 'LVM-'
+    uuid_len = 32
+
+    if (not dm_uuid.startswith(lvm_pfix)) or (len(dm_uuid) != (len(lvm_pfix) + (uuid_len * 2))):
```
Comment (jgrund) [inline chroma-agent/chroma_agent/device_plugins/linux_components/block_devices.py]: What case is there a string that starts with `LVM-` but doens't have the correct length?

File: chroma-agent/chroma_agent/device_plugins/linux_components/device_mapper.py [hunk 3]
```diff
@@ -251,7 +250,8 @@ def _read_mpath_partition(block_device, parent_mpath_name):
 
                 # Add this devices to the canonical path list.
                 for device in devices:
```
Comment (jgrund) [inline chroma-agent/chroma_agent/device_plugins/linux_components/device_mapper.py]: Does this file need to exist anymore?

File: chroma-agent/chroma_agent/device_plugins/linux_components/block_devices.py [hunk 4]
```diff
@@ -134,33 +141,142 @@ def fetch_device_list():
...
-def add_to_ndp(xs, ys):
-    for x in xs:
-        for y in ys:
-            ndp.add_normalized_device(x, y)
+def parse_lvm_uuids(dm_uuid):
+    """
+    :param dm_uuid: device mapper uuid string (combined vg and lv with lvm prefix) from device_scanner
+    :return: tuple of vg and lv UUIDs
+    """
+    lvm_pfix = 'LVM-'
```
Comment (jgrund) [inline chroma-agent/chroma_agent/device_plugins/linux_components/block_devices.py]: How reliable is parsing `DM_UUID` to get this info?

Comment (tanabarr) [reply chroma-agent/chroma_agent/device_plugins/linux_components/block_devices.py]: seems to be reliable that uuids are concatenated always for lvm devices in this manner (prefix-vguuid/lvuuid) and always 24 characters each

File: chroma-agent/chroma_agent/device_plugins/linux_components/block_devices.py (hunk 2 above)
Comment (tanabarr) [reply chroma-agent/chroma_agent/device_plugins/linux_components/block_devices.py]: should never happen, will amend

File: chroma-agent/chroma_agent/device_plugins/linux_components/block_devices.py (hunk 1 above)
Comment (tanabarr) [reply chroma-agent/chroma_agent/device_plugins/linux_components/block_devices.py]: when the device is not related to a device mapper

File: chroma-agent/chroma_agent/device_plugins/linux_components/device_mapper.py (hunk 3 above)
Comment (jgrund) [reply chroma-agent/chroma_agent/device_plugins/linux_components/device_mapper.py]: Is it just for tests or the actual source uses them?

If just the tests, then we should refactor to eliminate their usage in this patch.

Comment (tanabarr) [reply chroma-agent/chroma_agent/device_plugins/linux_components/device_mapper.py]: the source for those modules use device_mapper.py (and the output from dmsetup table command)

**Comment by tanabarr** [comment]
patch relies on https://github.com/intel-hpdd/device-scanner/pull/26/ to provide relevant records from the device scanner

**Comment by tanabarr** [comment]
http://jenkins.lotus.hpdd.lab.intel.com/job/integration-tests-shared-storage-configuration/2540/
pass

**Comment by jgrund** [comment]
Can you update the commit message to reflect the changes happening here?

**Pull request closed without merging**